File: src/main/java/com/java/scogen/dataGenerator/EmployeeFinanceDataset.java
package com.java.scogen.dataGenerator;

import com.java.scogen.pojo.EmployeeFinancePojo;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.util.List;
import java.util.stream.Collectors;

public class EmployeeFinanceDataset {

    private SparkSession spark;

    public EmployeeFinanceDataset(SparkSession spark) {
        this.spark= spark;
    }

    public Dataset<Row> employeeFinanceData(List<String> empNames)
    {
       return spark.createDataFrame(empNames.stream().map(EmployeeFinancePojo::new)
                .collect(Collectors.toList()),EmployeeFinancePojo.class);
    }

}

File: src/main/scala/com/scala/scogen/datagenerator/EmployeeFinanceDataset.scala
package com.scala.scogen.datagenerator

import com.scala.scogen.caseclasses.EmployeeFinancePojo
import org.apache.spark.sql.{Dataset, Row, SparkSession}

import scala.util.Random

class EmployeeFinanceDataset {

  def employeeFinanceData(spark:SparkSession,empIds: List[String]): Dataset[Row] = {
    import spark.implicits._
    val empFinancelist = empIds.map(empid=> {
      val ctc = new Random().nextInt(91) * 1000 + 10000
      EmployeeFinancePojo(empid,ctc,ctc*0.2,ctc*0.1,ctc*0.05)
    })

    spark.sparkContext.parallelize(empFinancelist).toDF()
  }

}
